library/include/ck/library/utility/check_err.hpp

```diff
@@ -23,6 +23,85 @@
+double get_relative_threshold()
+{
+    using F8   = ck::f8_t;
+    using F16  = ck::half_t;
+    using BF16 = ck::bhalf_t;
+    using F32  = float;
+    using I8   = int8_t;
+    using I32  = int32_t;
+
+    static_assert(is_same_v<ComputeDataType, F8> || is_same_v<ComputeDataType, F16> ||
+                      is_same_v<ComputeDataType, BF16> || is_same_v<ComputeDataType, F32> ||
+                      is_same_v<ComputeDataType, I8> || is_same_v<ComputeDataType, I32> ||
+                      is_same_v<ComputeDataType, int>,
+                  "Warning: Unhandled ComputeDataType for setting up the relative threshold!");
+    int compute_mantissa = 0;
+    if constexpr(is_same_v<ComputeDataType, I8> || is_same_v<ComputeDataType, I32> ||
+                 is_same_v<ComputeDataType, int>)
+    {
+        compute_mantissa = std::numeric_limits<ComputeDataType>::digits - 1;
+    }
+    else
+    {
+        compute_mantissa = NumericUtils<ComputeDataType>::mant;
+    }
+
+    static_assert(is_same_v<OutDataType, F8> || is_same_v<OutDataType, F16> ||
+                      is_same_v<OutDataType, BF16> || is_same_v<OutDataType, F32> ||
+                      is_same_v<OutDataType, I8> || is_same_v<OutDataType, I32> ||
+                      is_same_v<OutDataType, int>,
+                  "Warning: Unhandled OutDataType for setting up the relative threshold!");
+    int output_mantissa = 0;
+    if constexpr(is_same_v<OutDataType, I8> || is_same_v<OutDataType, I32> ||
+                 is_same_v<OutDataType, int>)
+    {
+        output_mantissa = std::numeric_limits<OutDataType>::digits - 1;
+    }
+    else
+    {
+        output_mantissa = NumericUtils<OutDataType>::mant;
+    }
+
+    int mantissa = (compute_mantissa < output_mantissa ? compute_mantissa : output_mantissa);
+
+    return std::pow(2, -mantissa) * 0.5;
+}
+
+template <typename ComputeDataType>
+double get_absolute_threshold(const double max_possible_num)
+{
+    using F8   = ck::f8_t;
+    using F16  = ck::half_t;
+    using BF16 = ck::bhalf_t;
+    using F32  = float;
+    using I8   = int8_t;
+    using I32  = int32_t;
+
+    static_assert(is_same_v<ComputeDataType, F8> || is_same_v<ComputeDataType, F16> ||
+                      is_same_v<ComputeDataType, BF16> || is_same_v<ComputeDataType, F32> ||
+                      is_same_v<ComputeDataType, I8> || is_same_v<ComputeDataType, I32> ||
+                      is_same_v<ComputeDataType, int>,
+                  "Warning: Unhandled ComputeDataType for setting up the relative threshold!");
+
+    auto expo = std::log2(std::abs(max_possible_num));
+
+    int mantissa = 0;
+    if constexpr(is_same_v<ComputeDataType, I8> || is_same_v<ComputeDataType, I32> ||
+                 is_same_v<ComputeDataType, int>)
+    {
+        mantissa = std::numeric_limits<ComputeDataType>::digits - 1;
+    }
+    else
+    {
+        mantissa = NumericUtils<ComputeDataType>::mant;
+    }
+
+    return 0.5 * std::pow(2, expo - mantissa);
```

Comment:
You didnt take into account AccDataType * NumberOfAccumulations. Then You have to get the max from these 3 errors.